torch/csrc/jit/passes/onnx/fixup_onnx_controlflow.cpp

```diff
@@ -258,25 +259,41 @@ bool IsUninitializedNode(Node* n) {
 
 // Infer shape and type of the uninitialized_output from the corresponding
 // output of the other subblock. prim::Uninitialized node is proven to be
-// unused. So replace this node with a constant of the inferred shape and type.
+// unused. So replace this node with a constant(TensorType) or
+// Sequence(ListType) of the inferred shape and type.
 void InferShapeTypeForUninitializedOutput(
     Graph* graph,
     Block* block,
     Value* uninitialized_output,
     Value* other_output) {
-  auto output_type = other_output->type()->expect<TensorType>();
-  auto elem_type = at::initialTensorOptions().dtype(output_type->scalarType());
-  Node* const_node = graph->create(::c10::onnx::Constant, 1);
-
-  if (output_type->sizes().concrete_sizes().has_value()) {
-    auto size = output_type->sizes().concrete_sizes().value();
-    const_node->t_(attr::value, at::zeros(size, elem_type));
-    const_node->output()->setType(other_output->type());
-    const_node->output()->copyMetadata(other_output);
-  } else {
-    const_node->t_(attr::value, at::zeros({}, elem_type));
-    const_node->output()->setType(
-        TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+  Node* const_node;
+  if (other_output->type()->cast<TensorType>() != nullptr) {
+    auto output_type = other_output->type()->expect<TensorType>();
+    auto elem_type =
+        at::initialTensorOptions().dtype(output_type->scalarType());
+    const_node = graph->create(::c10::onnx::Constant, 1);
+
+    if (output_type->sizes().concrete_sizes().has_value()) {
+      auto size = output_type->sizes().concrete_sizes().value();
+      const_node->t_(attr::value, at::zeros(size, elem_type));
+      const_node->output()->setType(other_output->type());
+    } else {
+      const_node->t_(attr::value, at::zeros({}, elem_type));
+      const_node->output()->setType(
+          TensorType::create(*(output_type->scalarType()), at::kCPU, {}, {}));
+    }
+  } else if (other_output->type()->cast<ListType>() != nullptr) {
+    auto output_type = other_output->type()->expect<ListType>();
+    TypePtr elem = output_type->getElementType();
+    const_node = graph->create(::c10::onnx::SequenceEmpty, 1);
+    if (elem->cast<TensorType>()->scalarType().has_value()) {
+      auto scalar_type = elem->cast<TensorType>()->scalarType().value();
+      auto onnx_type = ATenTypeToOnnxType(scalar_type);
+      const_node->i_(attr::dtype, onnx_type);
+      const_node->output()->setType(ListType::create(elem->cast<TensorType>()));
```

Comment:
this will copy over tensor shape from elem as well. is this intended?

Reply:
I thought that we copy over tensor shape from elem. I'm not particularly sure if that makes sense.
Is that more reasonable to use `setType(other_output->type())`?

Reply:
The current code is the same as `setType(other_output->type())`. `type` contains both `shape` and `scalarType`. If you intend to copy over both, then use `setType`, if you want to only copy `scalarType`, then you will need to update L#292